src/components/ConnectMetamask.jsx

```diff
@@ -21,11 +23,17 @@ init({
 export default function ConnectMetamask() {
 
     const [{ wallet, connecting }, connect, disconnect] = useConnectWallet();
+    const dispatch = useDispatch();
 
     useEffect(() => {
         if (!window.ethereum) {

```

Comment:
Here maybe we don't need the !window.ethereum check (There is possibility for useConnectWallet to be handling this). Check if we need it and if not remove